joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/datastructures/ProgramSummary.scala

```diff
@@ -122,22 +126,13 @@ trait TypedScope[M <: MethodLike, F <: FieldLike, T <: TypeLike[M, F]](summary:
     tag: ClassTag[M]
   ): Option[M] = typeFullName match {
     case None =>
-      // This function uses the `implicit tag` (IntelliJ incorrectly marks it as unused)
-      def matchingM: PartialFunction[MemberLike, M] = { case m: M if m.name == callName => m }
       // TODO: The typesInScope part is to imprecisely solve the unimplemented polymorphism limitation
-      membersInScope.collectFirst(matchingM).orElse { typesInScope.flatMap(_.methods).collectFirst(matchingM) }
-    case Some(tfn) =>
-      val methodsWithEqualArgs = tryResolveTypeReference(tfn).flatMap { t =>
-        Option(
-          t.methods.filter(m => m.name == callName && m.parameterTypes.filterNot(_._1 == "this").size == argTypes.size)
-        )
+      membersInScope.collectFirst(matchingM(callName)).orElse {
```

Comment:
Could we give `argTypes` a default so that nonoverloaded languages can ignore this? E.g. `argTypes: List[String] = Nil`